src/engine/duckduckgo.rs

```diff
@@ -40,7 +42,21 @@ pub fn extract_links(page: &str) -> Option<Vec<String>> {
     let target_elements = doc.find(Class("result__a"));
     let links: Vec<String> = target_elements
         .filter_map(|node| node.attr("href"))
-        .map(|link| String::from(link))
+        .map(|link| {
```

Comment:
Using `filter_map` instead of `map` can be better.
Normally we don't want the program panic if `uddg` don't exists in *redirect link*.  For this un-common scenario, I'd like to filter the link rather than make the whole program panic.